src/Extension/HeadingPermalink/HeadingPermalinkProcessor.php

```diff
@@ -45,27 +45,40 @@ public function setEnvironment(EnvironmentInterface $environment): void
 
     public function __invoke(DocumentParsedEvent $e): void
     {
-        $min = (int) $this->config->get('heading_permalink/min_heading_level');
-        $max = (int) $this->config->get('heading_permalink/max_heading_level');
-
-        $slugLength = (int) $this->config->get('slug_normalizer/max_length');
+        $min           = (int) $this->config->get('heading_permalink/min_heading_level');
+        $max           = (int) $this->config->get('heading_permalink/max_heading_level');
+        $attachHeading = (bool) $this->config->get('heading_permalink/attach_heading');
+        $idPrefix      = (string) $this->config->get('heading_permalink/id_prefix');
+        $slugLength    = (int) $this->config->get('slug_normalizer/max_length');
+        $headingClass  = $this->config->get('heading_permalink/heading_class');
+
+        if ($idPrefix !== '') {
+            $idPrefix .= '-';
+        }
 
         foreach ($e->getDocument()->iterator(NodeIterator::FLAG_BLOCKS_ONLY) as $node) {
             if ($node instanceof Heading && $node->getLevel() >= $min && $node->getLevel() <= $max) {
-                $this->addHeadingLink($node, $slugLength);
+                $this->addHeadingLink($node, $slugLength, $idPrefix, $attachHeading, $headingClass);
             }
         }
     }
 
-    private function addHeadingLink(Heading $heading, int $slugLength): void
+    private function addHeadingLink(Heading $heading, int $slugLength, string $idPrefix, bool $attachHeading, string $headingClass): void
     {
         $text = StringContainerHelper::getChildText($heading, [RawMarkupContainerInterface::class]);
         $slug = $this->slugNormalizer->normalize($text, [
             'node' => $heading,
             'length' => $slugLength,
         ]);
 
-        $headingLinkAnchor = new HeadingPermalink($slug);
+        if ($attachHeading) {
+            $heading->data->set('attributes/id', $idPrefix . $slug);
+            if ($headingClass !== '') {
+                $heading->data->append('attributes/class', $this->config->get('heading_permalink/heading_class'));
```

Comment:
I think we should allow the heading class to always be configurable - even when they aren't using the new `attach_heading` config option.  (We can use the documentation to explain that the `heading_class` option is especially helpful when the `attach_heading` behavior is enabled, but that it can still be used more generally if desired)

Reply:
I am always for more options and flexibility :)